importlib_metadata/__init__.py

```diff
@@ -473,7 +473,7 @@ def search(self, name):
         for child in self.children():
             n_low = child.lower()
             if (n_low in name.exact_matches
-                    or n_low.startswith(name.prefix)
+                    or n_low.replace('.', '_').startswith(name.prefix)
```

Comment:
Doesn't this also need to account for other character substitutions in normalization?

Reply:
I see now this is to support legacy behaviors where egg-info and dist-info have dots.